src/main.rs

```diff
@@ -70,6 +67,77 @@ impl BFInterp {
             MetadataOperand => todo!(),
         }
     }
+    
+    fn load(&mut self, op: &Operand, dest: &llvm_ir::name::Name) {
+        match op {
+            LocalOperand { name, .. } => {
+                self.mem.insert(dest.clone(), *self.mem.get(name).unwrap());
+                self.mem.remove(name);
+            },
+            ConstantOperand(_) => todo!(),
+            MetadataOperand => todo!(),
+        }
+    }
+
+    fn call_func(&mut self, call: &Call) { 
+        let arg;    
+        match &call.arguments[1].0 {
+            LocalOperand { name, .. } => {
+                arg = self.mem.get(name).unwrap();
+            },
+            ConstantOperand(_) => todo!(),
+            MetadataOperand => todo!(),
+        }
+        
+        match &call.function {
+            Left(_) => todo!(),
+            Right(op) => {
+                match op {
+                    LocalOperand { .. } => todo!(),
+                    ConstantOperand(con_op) => {
+                        match con_op.as_ref() {
+                            Constant::GlobalReference { name, .. } => {
+                                match name {
+                                    Name(n) => {
+                                        if n.as_str() == "printf" {
+                                            println!("{}", arg);
+                                        }
+                                        // match self.find_symbol(n.as_str()) {
+                                        //     Some(c_void) => {
+                                        //         //println!("{:?}", c_void);
+                                        //         let func_pointer = c_void.cast::<fn()>();
+                                        //     },
+                                        //     None => todo!(),
+                                        // }
+                                    },
+                                    Number(_) => todo!(),
+                                }
+                            },
+                            _ => todo!(),
+                        }
+                    },
+                    MetadataOperand => todo!(),
+                }
+            },
+        }
+    }
+
+    fn find_symbol(&mut self, sym: &str) -> Option<*mut c_void> {
+        use std::ffi::CString;
+
+        let sym_arg = CString::new(sym).unwrap();
+        let addr = unsafe { dlsym(RTLD_DEFAULT, sym_arg.into_raw()) };
```

Comment:
`into_raw()` will leak memory here, because:

> Consumes the CString and transfers ownership of the string to a C caller.
>
> The pointer which this function returns must be returned to Rust and reconstituted using CString::from_raw to be properly deallocated. 

I think you want [as_ptr()](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.as_ptr] which means that your code retains ownership and the `CString` is freed when it goes out of scope.

Reply:
I don't think we still need this code as we won't be doing any external calls soon (due to the complexity of it as discussed offline).